hw/ip/aes/doc/dv/index.md

```diff
@@ -60,22 +64,60 @@ It can be created manually by invoking [`regtool`]({{< relref "util/reggen/READM
-
-#### Functional coverage
+`aes_base_vseq` provides commonly used handles, variables, functions and tasks that the test sequences can simply use / call.
+The tasks can be split into two groups and those that provide more complex functionality.
+Simple tasks include:
+* **aes_init**:    Initialize the AES module from the randomized environment variables in the config.
+* **set_op**:      Set AES operation to encrypt or decrypt.
+* **write_key**:   Write initial key to AES init key registers.
+* **add_data**:    Add the next 128 block to the input registers.
+* **read_output**:  Poll the status register for data ready bit and read the result from AES output registers.
+* **clear_reg**:  Based on the input this function clears data input-, data output- or key-registers or any combination of these.
+* **set_manual_trigger**: Chooses between AES auto start and manual start.
+* **trigger_start**:  Set the start bit to trigger a new encryption/decryption.
+
+More complex tasks include:
+These are the ones used by the higher level sequences and the ones that should be used to create new tests from:
+* **generate_message_queue**: Generate a queue of randomized message items. 
+Each item will describe the parameters of a message but not hold any data, the data will be added later.
+This function does not call any sub-functions.
+* **send_msg_queue**: Take the queue of messages items and process them one by one.
+Send_msg_queue converts each message item into a queue of message transactions each called an aes_items by calling generate aes_item_queue().
+Then each message now described by a queue of items is processed by calling send_msg().
+* **generate_aes_item_queue**: Expands a message into a queue of a configuration item and N Data items where N = Message_length/block size.
+Data is randomized based on the constraints.
+* **send_msg**: Take a queue of configuration and data items and pass them to the AES for processing.
+This task will handle configuration of the core based on the configuration item.
+Then based on the test configuration it will do either a balanced or unbalanced processing of the data items.
+A balanced processing implies that for each input the task will wait until the resulting output have been read before attempting to write the next input.
+The unbalanced version will write the next data item as soon as the AES status register indicates that a new input will be accepted.
+In unbalanced mode the send_msg task also handles reading the output register.
+Knobs are available to control the balance between reads and writes.
+In either case the status_fsm task is called.
+* **status_fsm**: Read the status of the AES IP and based on the inputs return the status.
+When the task is in control of reading the output register it will poll the status until it indicates that the output is valid.
+It will also track the progress of processing detecting if something has gone differently than expected.
```

Comment:
Thanks for joining the sentences. As a nit, you probably want a comma between "processing" and "detecting", otherwise a reader can think you're talking about processing something that's called "detecting" and get a bit tripped up.

Reply:
fixed